src/components/OntologyBrowser/SelectionTree.tsx

```diff
@@ -78,30 +79,66 @@ export class SelectionTree extends Component<SelectionTreeProps, SelectionTreeSt
     selectedKeys.indexOf(eventKey.toString()) !== -1;
 
   onChange = (e: React.ChangeEvent<HTMLInputElement>, treeData: TreeNode[]) => {
-    treeData.forEach(node => this.searchInTree(e.target.value, node));
-    this.setState({ treeData });
+    const hits: string[] = []
+
+    if(e.target.value.length >= MIN_SEARCH_TEXT_LENGTH){
```

Comment:
I don't know what you think, but I think because the if/else don't returned, seting the state at the end make the code more consise

```javascript
    const hits: string[] = []
    let newExpandedKeys: string[] = []

    if(e.target.value.length >= MIN_SEARCH_TEXT_LENGTH){
      treeData
        .forEach(node => this.searchInTree(e.target.value, node, hits))
      newExpandedKeys = hits;
    } else {
      treeData.forEach(node => this.showAll(node))
      newExpandedKeys = getInitialKeysForExpand(treeData)
    }

    this.setState({
      treeData: treeData,
      expandedKeys: getInitialKeysForExpand(treeData),
    });
```
its up to you